src/internals.h

```diff
@@ -35,6 +36,26 @@ namespace task {
  */
 extern const gul14::string_view abort_marker;
 
+enum class ErrorCause { terminated_by_script, aborted, uncaught_error };
+
+/**
+ * Remove abort markers from the given error message and determine the cause of the error.
+ *
+ * All abort markers are removed from the given error message. If no marker was present,
+ * the message describes a "normal" error and ErrorCause::uncaught_error is returned.
+ *
+ * If the message contains at least two abort markers and there is no message enclosed
+ * between the first two of them, this is considered an explicit termination request by
+ * the running script. The error message is set to a corresponding explanation ("Script
+ * called terminate_sequence()") and ErrorCause::terminated_by_script is returned.
+ *
+ * If the error message contains at least one abort marker and an error message, it is
+ * considered an abort request (either by the user or by timeouts) and ErrorCause::aborted
+ * is returned. If there are at least two abort markers, the error message is set to the
+ * text enclosed by the first two of them.
```

Comment:
The explanation also shows how (too) complicated that is. Why is there two times the phrase `If there are at least two markers`? Simplify code and explanation?

Reply:
You are right. It was basically unreadable. Have a look at the new one.